From bc05c36cbe8cafb7a6d9ea3d4a10cfac5bc0d532 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <rymg19@gmail.com>
Date: Tue, 20 Jul 2021 14:18:31 -0500
Subject: [PATCH] Enable support for libfdk-aac and OpenH264

---
 BUILD.gn | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/third_party/ffmpeg/BUILD.gn b/third_party/ffmpeg/BUILD.gn
index 0ac39bfbd0..309d0ac870 100644
--- a/third_party/ffmpeg/BUILD.gn
+++ b/third_party/ffmpeg/BUILD.gn
@@ -440,6 +440,11 @@ if (is_component_ffmpeg) {
         "-L",
         rebase_path(target_gen_dir, root_build_dir),
       ]
+
+      libs = [
+        "fdk-aac",
+        "openh264",
+      ]
     } else if (is_win) {
       # Windows component build needs the .def file to export ffmpeg symbols.
       deps += [ ":ffmpeg_generate_def" ]
-- 
2.47.1

