From fc400fe2d47cbd8487e6251df99598f62b6d21d2 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <git@refi64.dev>
Date: Wed, 24 Jul 2024 17:55:22 -0500
Subject: [PATCH] Fix files in libavcodec/bsf/ failing to find headers

---
 BUILD.gn | 1 +
 1 file changed, 1 insertion(+)

diff --git a/third_party/ffmpeg/BUILD.gn b/third_party/ffmpeg/BUILD.gn
index 5107beb3d6..00dee3a141 100644
--- a/third_party/ffmpeg/BUILD.gn
+++ b/third_party/ffmpeg/BUILD.gn
@@ -214,6 +214,7 @@ target(link_target_type, "ffmpeg_internal") {
   include_dirs = [
     platform_config_root,
     ".",
+    "libavcodec",
   ]
   defines = [
     "HAVE_AV_CONFIG_H",
-- 
2.49.0

