From d6637ee51310441ec2db2306b92c1ba78d2fba59 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <rymg19@gmail.com>
Date: Wed, 31 Aug 2022 10:09:01 -0500
Subject: [PATCH] Clang build script: Don't build against the sysroot

---
 tools/clang/scripts/build.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/tools/clang/scripts/build.py b/tools/clang/scripts/build.py
index 7547aa3d102ba..cfea7aa0ab1d6 100755
--- a/tools/clang/scripts/build.py
+++ b/tools/clang/scripts/build.py
@@ -814,7 +814,7 @@ def main():
   ninja_dir = os.path.join(THIRD_PARTY_DIR, 'ninja')
   os.environ['PATH'] = ninja_dir + os.pathsep + os.environ.get('PATH', '')
 
-  if sys.platform.startswith('linux'):
+  if sys.platform.startswith('linux') and False:
     with timer.time('get sysroots'):
       sysroot_amd64 = DownloadDebianSysroot('amd64', args.skip_checkout)
       sysroot_i386 = DownloadDebianSysroot('i386', args.skip_checkout)
@@ -920,7 +920,7 @@ def main():
     if sys.platform.startswith('linux'):
       base_cmake_args += [ '-DLLVM_STATIC_LINK_CXX_STDLIB=ON' ]
 
-  if sys.platform.startswith('linux'):
+  if sys.platform.startswith('linux') and False:
     # Add the sysroot to base_cmake_args.
     if platform.machine() == 'aarch64':
       base_cmake_args.append('-DCMAKE_SYSROOT=' + sysroot_arm64)
@@ -1251,7 +1251,7 @@ def main():
   if sys.platform.startswith('linux'):
     runtimes_triples_args['i386-unknown-linux-gnu'] = {
         "args": [
-            'CMAKE_SYSROOT=%s' % sysroot_i386,
+            # 'CMAKE_SYSROOT=%s' % sysroot_i386,
             # TODO(crbug.com/40242553): pass proper flags to i386 tests so they
             # compile correctly
             'LLVM_INCLUDE_TESTS=OFF',
@@ -1263,7 +1263,7 @@ def main():
     }
     runtimes_triples_args['x86_64-unknown-linux-gnu'] = {
         "args": [
-            'CMAKE_SYSROOT=%s' % sysroot_amd64,
+            # 'CMAKE_SYSROOT=%s' % sysroot_amd64,
         ],
         "profile": True,
         "sanitizers": True,
@@ -1277,7 +1277,7 @@ def main():
     # Maybe it should work for builtins too?
     runtimes_triples_args['armv7-unknown-linux-gnueabihf'] = {
         "args": [
-            'CMAKE_SYSROOT=%s' % sysroot_arm,
+            # 'CMAKE_SYSROOT=%s' % sysroot_arm,
             # Can't run tests on x86 host.
             'LLVM_INCLUDE_TESTS=OFF',
         ],
@@ -1288,7 +1288,7 @@ def main():
     }
     runtimes_triples_args['aarch64-unknown-linux-gnu'] = {
         "args": [
-            'CMAKE_SYSROOT=%s' % sysroot_arm64,
+            # 'CMAKE_SYSROOT=%s' % sysroot_arm64,
             # Can't run tests on x86 host.
             'LLVM_INCLUDE_TESTS=OFF',
         ],
-- 
2.49.0

